modules/ztp-image-based-upgrade-concept.adoc

```diff
@@ -0,0 +1,132 @@
+[cols=3*, width="100%", options="header"]
+|====
+|All actions share same strategy
+|Some actions share same strategy
+|All action different strategy
+
+a|[source,yaml]
+----
+plan:
+- actions: ["Prep", "Upgrade", "FinalizeUpgrade"]
+  rolloutStrategy:
+    maxConcurrency: 200
+    timeout: 60
+----
+a|[source,yaml]
+----
+plan:
+- actions: ["Prep", "Upgrade"]
+  rolloutStrategy:
+    maxConcurrency: 200
+    timeout: 60
+- actions: ["FinalizeUpgrade"]
+  rolloutStrategy:
+    maxConcurrency: 500
+    timeout: 10
+----
+a|[source,yaml]
+----
+plan:
+- actions: ["Prep"]
+  rolloutStrategy:
+    maxConcurrency: 200
+    timeout: 60
+- actions: ["Upgrade"]
+  rolloutStrategy:
+    maxConcurrency: 200
+    timeout: 20
+- actions: ["FinalizeUpgrade"]
+  rolloutStrategy:
+    maxConcurrency: 500
+    timeout: 10
+----
+|====
+
+[IMPORTANT]
+====
+The clusters that fail one of the actions will skip the rest of the actions in the same step.
+====
+
+The `ImageBasedGroupUpgrade` API accepts the following actions:
+
+`Prep`:: Start preparing the upgrade resources by moving to the `Prep` stage.
+`Upgrade`:: Start the upgrade by moving to the `Upgrade` stage.
+`FinalizeUpgrade`:: Finalize the upgrade on selected clusters that completed the `Upgrade` action by moving to the `Idle` stage.
+`Rollback`:: Start a rollback only on successfully upgraded clusters by moving to the `Rollback` stage.
+`FinalizeRollback`:: Finalize the rollback by moving to the `Idle` stage.
+`AbortOnFailure`:: Cancel the upgrade on selected clusters that failed the `Prep` or `Upgrade` actions by moving to the `Idle` stage.
+`Abort`:: Cancel an ongoing upgrade only on clusters that are not upgraded yet by moving to the `Idle` stage.
+
+The following action combinations are supported, where one pair of brackets signify one step in the `plan` section:
+
+* `["Prep"]`, `["Abort"]`
+* `["Prep", "Upgrade", "FinalizeUpgrade"]`
+* `["Prep"]`, `["AbortOnFailure"]`, `["Upgrade"]`, `["AbortOnFailure"]`, `["FinalizeUpgrade"]`
+* `["Rollback", "FinalizeRollback"]`
+
+The following combinations are for cases where you need to resume or cancel an ongoing upgrade from a completely new `ImageBasedGroupUpgrade` CR:
+
+* `["Upgrade","FinalizeUpgrade"]`
+* `["FinalizeUpgrade"]`
+* `["FinalizeRollback"]`
+* `["Abort"]`
+* `["AbortOnFailure"]`
+
+[id="ztp-image-based-upgrade-cluster-labeling_{context}"]
+== Labeling for cluster selection
+
+You can use the `clusterLabelSelectors` field for initial cluster selection.
+In addition, {cgu-operator} labels the managed clusters according to the results of their last stage transition.
+
+If a stage completes or fails, {cgu-operator} marks the relevant clusters with the following labels:
+
+* `lcm.openshift.io/ibgu-<stage>-completed`
+* `lcm.openshift.io/ibgu-<stage>-failed`
+
+For example, if you want to cancel the upgrade for all managed clusters except for clusters that successfully completed the upgrade, you can append an `Abort` action to your plan.
```

Comment:
I don't quite understand the transition and flow of the docs here. The example seems to come out of nowhere. Is there something that can be added that makes the transition from the list of labels to an example of adding the `Abort` action make more sense?